cmd/server/main.go

```diff
@@ -41,6 +41,7 @@ import (
 	"go.temporal.io/server/common/log/tag"
 	_ "go.temporal.io/server/common/persistence/sql/sqlplugin/mysql"      // needed to load mysql plugin
 	_ "go.temporal.io/server/common/persistence/sql/sqlplugin/postgresql" // needed to load postgresql plugin
+	_ "go.temporal.io/server/common/persistence/sql/sqlplugin/sqlite"     // needed to load sqlite plugin
```

Comment:
This is what breaks the build. Apparently w/o this line, `sqlite` package is not even compiled. With this line it gets compiled but compilation fails because we don't use CGO for our production Linux builds and 
> ***Important: because this is a `CGO` enabled package you are required to set the environment variable `CGO_ENABLED=1` and have a `gcc` compile present within your path.***

Reply:
I don't want to enable CGO and don't know what is the best way to solve this. The only [one place](https://github.com/temporalio/temporal/blob/64ef4e223785fd5ba77c51de9101157e53092820/common/persistence/sql/sqlplugin/sqlite/driver.go#L38-L39) where `sqlite3` is used. Maybe we can avoid importing this package and use some other approach.

Reply:
Well, you already have a driver which utilizes the SQLite CGO package. The alternative is to use a different type of embedded database... or something like this - https://www.reddit.com/r/golang/comments/n1u2b1/new_advanced_cgofree_sqlite_package/

I agree about avoiding CGO, maybe create a different build pipeline? Or it's too complicated?

Reply:
I can also turn off the plugin in server and SQL so DataDog can patch their TemporalLite server.

Reply:
FYI there was lots of discussion around CGO on the original PR. I did have a non-CGO version working at one point using `modernc.org/sqlite`, but for now we decided to skip importing the driver by default: https://github.com/temporalio/temporal/pull/1968#issuecomment-930435029

I'd suggest doing the same here and follow up on supporting non-CGO builds or dealing with cross-compilation issues introduced by CGO separately.

Reply:
In the mean time it would be pretty straightforward to add a file to the `main` package which uses a build tag to conditionally import the sqlite driver when CGO is enabled. That way anyone wanting to try out the `sqlite` driver just needs to compile from source rather than needing to write their own Temporal server wrapper or pull in Temporalite.